### DISCUSSION TITLE
SDKs could bundle and release backend error codes

### DISCUSSION THREAD

**Discussion by craicoverflow** [redhat-developer/app-services-sdk-go#269] [Ideas]
The custom error codes returned by the APIs could be constantly evolving as new features get added.

There is a possibility that SDK consumers are unaware of new error codes and cannot handle them appropriately.

It would be great if we could obtain the error codes from backends and release them with the SDKs.

The main blocker for this I see is there is no straightforward way to do this right now - unless the API teams provide micro-packages that expose these error codes.

**Comment by wtrocki** [comment]
I have discussed that in context of registry. We could introduce enum with error codes that is used in examples etc. 
This enum will not cause any breaking changes in API but it will need to be maintained in openapi file. 

Additionally that will be also easy to enforce on spectral side?

**Comment by wtrocki** [comment]
> - unless the API teams provide micro-packages that expose these error codes.

They do actually specify all error codes in single files (for all 3 services I have worked with doing so)

**Reply by craicoverflow** [reply]
Yes they do. Two problems is that they are still private and so cannot be imported.

Second is that we would want them available in Go, Java, JS

**Reply by wtrocki** [reply]
Yes. See my comment of having them as enum fields in the OpenAPI spec. then codegen will generate them.

**Reply by craicoverflow** [reply]
That would be good - worthy of an experiment :)

**Reply by wtrocki** [reply]
We have done enum definitions in SRS API for status codes that work great. We can take action and try this in kafka first
